code/game/machinery/atmos_exporter.dm

```diff
@@ -0,0 +1,293 @@
+	if(..())
+		return
+	switch(href_list["action"])
+		if ("choose")
+			reset()
+			export_target = input(usr, "Choose your export quality", "Quality:") as null|anything in quality
+			if (!src || QDELETED(src))
+				return
+			switch(export_target)
+				if("Economy")
+					export_name = economy_name
+					gas = e_gas.Copy()
+					upressure = e_pressure + 200
+					lpressure = e_pressure - 200
+					utemp = e_temp + 50
+					ltemp = e_temp - 50
+					concentration += e_concentration1
+					price = e_price
+				if("Standard")
+					export_name = standard_name
+					gas = s_gas.Copy()
+					upressure = s_pressure + 100
+					lpressure = s_pressure - 100
+					utemp = s_temp + 30
+					ltemp = s_temp - 30
+					concentration += s_concentration1
+					concentration += s_concentration2
+					price = s_price
+				if("Premium")
+					export_name = premium_name
+					gas = p_gas.Copy()
+					upressure = s_pressure + 50
+					lpressure = s_pressure - 50
+					utemp = s_temp + 15
+					ltemp = s_temp - 15
+					concentration += p_concentration1
+					concentration += p_concentration2
+					concentration += p_concentration3
+					price = p_price
+			updateUsrDialog()
+		if ("scan")
+			for(var/obj/machinery/portable_atmospherics/canister/C in get_turf(can_turf))
+				atmosanalyzer_scan(C.air_contents, usr, C)
+				playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 100, 1)
+		if ("send")
+			for(var/obj/machinery/portable_atmospherics/canister/C in get_turf(can_turf))
+				if((lpressure) <= C.air_contents.return_pressure() && C.air_contents.return_pressure() <= (upressure))
+					if((ltemp) <= C.air_contents.temperature && C.air_contents.temperature <= (utemp))
+						var/total_moles = C.air_contents.total_moles()
+						var/list/can_gas = C.air_contents.gases
+						var/i = 0
+						for(var/id in gas)
+							if(can_gas[id])
+								var/gas_concentration = (can_gas[id][MOLES]/total_moles)*100
+								i++
+								if(gas_concentration <= (concentration[i]-5) || gas_concentration >= (concentration[i]+5))
+									visible_message("<span class='danger'>Invalid gas composition detected!</span>")
+									playsound(src, 'sound/machines/defib_failed.ogg', 100, 1)
+									return
+							else
+								visible_message("<span class='danger'>Invalid gas detected!</span>")
+								playsound(src, 'sound/machines/defib_failed.ogg', 100, 1)
+								return
+						visible_message("<span class='danger'>Valid [export_target] canister detected... exporting now. Your department will receive [price] points!</span>")
+						C.anchored = 1
+						for(var/obj/machinery/atmos_points_exporter/EX in atmos_exporter_list)
+							EX.ship()
+						for(var/obj/machinery/engi_points_manager/EPM in engi_points_list)
+							EPM.GBP += price
+							EPM.GBPearned += price
+						if(export_target == "premium")
+							visible_message("<span class='danger'>You have been blessed by the atmos gods for exporting a premium canister!</span>")

```

Comment:
I'm fine with the goal, I'm against the message (which is why I've commented on the message's line)